File: leetcode.com/0021_merge_two_sorted_lists/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        root = ListNode()
        curr = root

        while l1 and l2:
            if l1.val < l2.val:
                curr.next = ListNode(l1.val)
                curr = curr.next
                l1 = l1.next
            else:
                curr.next = ListNode(l2.val)
                curr = curr.next
                l2 = l2.next

        l = l1 or l2
        while l:
            curr.next = ListNode(l.val)
            curr = curr.next

            l = l.next

        return root.next

File: leetcode.com/0021_merge_two_sorted_lists/solution.c
struct ListNode *mergeTwoLists(struct ListNode *l1, struct ListNode *l2)
{
    if (l1 && !l2)
        return l1;

    if (!l1 && l2)
        return l2;

    if (!l1 && !l2)
        return NULL;

    struct ListNode *head = malloc(sizeof(struct ListNode));
    struct ListNode *curr = head;

    while (l1 && l2) {
        if (l1->val < l2->val) {
            curr->val = l1->val;
            l1 = l1->next;
        } else {
            curr->val = l2->val;
            l2 = l2->next;
        }

        if (l1 && l2) {
            curr->next = malloc(sizeof(struct ListNode));
            curr = curr->next;
            curr->next = NULL;
            curr->val = -1;
        }
    }

    curr->next = (l1 == NULL) ? l2 : l1;
    return head;
}
